Pipeline instructions

pmatthews05 opened the pull request:
# [85](https://github.com/Azure/caf-terraform-landingzones-starter/issues/85)

## PR Checklist

---

<!-- Use the check list below to ensure your branch is ready for PR. -->

- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My code follows the code style of this project.
- [ ] I ran lint checks locally prior to submission.
- [ ] Have you checked to ensure there aren't other open Pull Requests for the same update/change?

## Description

Created a step by step instruction on deploying pipeline using sandpit. Ensuring a person following along would not require to much previous knowledge of how the project hangs together. 

Update to paths in pipelines to ensure it would work. 

## Does this introduce a breaking change

- [ ] YES
- [X] NO

<!-- If this introduces a breaking change, please describe the impact and migration path for existing applications below. -->

## Testing

Follow the instructions inside configuration/sandpit/pipelines/README-pipelines.md from beginning to end in a clean environment.

pmatthews05 committed:
Corrected sandpit pipeline example

pmatthews05 committed:
Tidy up & Format

pmatthews05 committed:
explain prefix

pmatthews05 committed:
Improvements how to clone from DevOps and check in

pmatthews05 committed:
Note to switch project name if different from default.

pmatthews05 committed:
mistake fixup

pmatthews05 committed:
Few amendments

pmatthews05 committed:
shortened keyvault name to ensure work

ghost commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/signed)](https://cla.opensource.microsoft.com/Azure/caf-terraform-landingzones-starter?pullRequest=86) <br/>All CLA requirements met.

pmatthews05 committed:
KVName and Perms

Shortened name to allow upto 9 letter prefix.
Added keyvault permission for MSI and AD Group

arnaudlh committed:
Merge branch 'starter' into pipelineInstructions

arnaudlh commented:
thanks @pmatthews05 for your contribution to CAF Terraform landing zones, sorry it took a bit of time to get in!
